File: src/main/java/com/gestion_actas/gestion/de/actas/services/PersonalService.java
package com.gestion_actas.gestion.de.actas.services;

import com.gestion_actas.gestion.de.actas.model.Personal;
import com.gestion_actas.gestion.de.actas.repository.PersonalRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
import org.springframework.stereotype.Service;

import java.util.Optional;

@Service
public class PersonalService {

    @Autowired
    private PersonalRepository personalRepository;

    private final BCryptPasswordEncoder passwordEncoder = new BCryptPasswordEncoder();

    public Personal registrarUsuario(Personal usuario) {
        usuario.setContrasena(passwordEncoder.encode(usuario.getContrasena()));
        return personalRepository.save(usuario);
    }

    public Optional<Personal> login(String dni, String contrasena) {
        Optional<Personal> userOpt = personalRepository.findByDni(dni);
        if (userOpt.isPresent()) {
            Personal user = userOpt.get();
            if (passwordEncoder.matches(contrasena, user.getContrasena())) {
                return Optional.of(user);
            }
        }
        return Optional.empty();
    }

    public Optional<Personal> buscarPorDni(String dni) {
        return personalRepository.findByDni(dni);
    }
}

File: BackendActasPHP/app/Services/PersonalService.php
<?php

namespace App\Services;

use App\Models\Personal;
use Illuminate\Support\Facades\Hash;

class PersonalService
{
    public function registrarUsuario($data)
    {
        $data['contrasena'] = Hash::make($data['contrasena']);
        return Personal::create($data);
    }

    public function login($dni, $contrasena)
    {
        $user = Personal::where('dni', $dni)->first();
        if ($user && Hash::check($contrasena, $user->contrasena)) {
            return $user;
        }
        return null;
    }

    public function buscarPorDni($dni)
    {
        return Personal::where('dni', $dni)->first();
    }
}